atc/api/accessor/handler.go

```diff
@@ -3,31 +3,39 @@ package accessor
 import (
 	"context"
 	"net/http"
+
+	"github.com/concourse/concourse/atc/auditor"
 )
 
 func NewHandler(
 	handler http.Handler,
 	accessFactory AccessFactory,
 	action string,
+	aud auditor.Auditor,
 ) http.Handler {
 	return accessorHandler{
 		handler:       handler,
 		accessFactory: accessFactory,
 		action:        action,
+		auditor:       aud,
 	}
 }
 
 type accessorHandler struct {
 	handler       http.Handler
 	accessFactory AccessFactory
 	action        string
+	auditor       auditor.Auditor
 }
 
 func (h accessorHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	acc := h.accessFactory.Create(r, h.action)
 	ctx := context.WithValue(r.Context(), "accessor", acc)
 
+	if acc != nil {
```

Comment:
This nil check shouldn't be necessary

Reply:
To explain this nil check, I found that the access object is actually a nil object on a variety of occasions.  For example, when an unauthenticated user is accessing the site, it technically has no access object.

in order to not throw errors in creating an audit log, I found that some level of nil checking was required to confirm that an access object existed.

I am however open to other solutions in order to handle this.

Reply:
It gets created on line 33. It will never be nil ;)

Also every request through the API goes through this handler. So every request will have a an `access` object, even unauthenticated users. Do you mean its nil in the tests?

Reply:
So, in an eating crow moment. I had remembered building this originally without a nil check on the handler and both the application and the tests were throwing errors when trying to run and perform audits on unauthenticated users.

However, as I was preparing a more in depth answer to your question, I can't reproduce the application errors I was seeing beforehand.  (which may mean that I either solved the issue, or am just remembering poorly).

It does look like the unit tests are throwing errors in both [the handler tests](https://github.com/loghen41/concourse/blob/auditor/atc/api/accessor/handler_test.go#L46) and throughout the [api test suite](https://github.com/loghen41/concourse/tree/auditor/atc/api) in general.

I'll remove this nil as it's performing correctly in the application, and adjust the unit tests accordingly.  Sorry to mis-remember this.

Reply:
@pivotal-jwinters I just pushed up a change updating all of the unit tests, and removing unneeded nil checks.  Please feel free to review when you get a moment and let me know if you see anything else I missed. 

I'll mark all of these conversations as resolved in the meantime.